### PR TITLE
[Go] change embedder plugins to use Init and New

### PR DESCRIPTION
**Pull request by ianlancetaylor**
Change the pinecone and localvec plugins to use a `New` function that
returns an `ai.Retriever`, and an `Init` function that registers one with genkit.

This makes the pinecone and localvec plugins consistent with each other,
and with other plugins such as googleai.

### CODE REVIEW

File: go/plugins/pinecone/genkit.go [hunk 1]
```diff
@@ -36,7 +36,19 @@ import (
 // documents in pinecone.
 const defaultTextKey = "_content"
 
-// Init registers actions to be used with ai.
+// Init registers an action to be used with ai.
```
Comment (randall77) [inline go/plugins/pinecone/genkit.go:39]: Same here.

**Comment by ssbushi** [comment]
Thanks Ian. 

AFAIU, we should use `New` if we want to use the retriever immediately (without lookup) and `Init` if we want to make it available within Genkit (eg for other plugins to lookup etc.)

Is that correct?

**Review by ssbushi — APPROVED** [review]

**Comment by ianlancetaylor** [comment]
@ssbushi Yes, that's correct.  We can change that if something else seems better, but let's at least be consistent for now.

File: go/plugins/localvec/localvec.go [hunk 2]
```diff
@@ -34,6 +34,16 @@ import (
 	"github.com/firebase/genkit/go/core/logger"
 )
 
+// Init register an action to be used with ai.
```
[the remark this answers is not in the document]
Comment (ianlancetaylor) [reply go/plugins/localvec/localvec.go:37]: Sorry, I don't quite understand what you are suggesting.  What would the whole comment look like?

[the remark this answers is not in the document]
Comment (randall77) [reply go/plugins/localvec/localvec.go:37]: "Init registers all the actions in this package with [ai]'s Register calls."

[the remark this answers is not in the document]
Comment (ianlancetaylor) [reply go/plugins/localvec/localvec.go:37]: Thanks, done.

File: go/plugins/pinecone/genkit.go [hunk 3]
```diff
@@ -36,7 +36,19 @@ import (
...
+		return err
+	}
+	ai.RegisterRetriever("pinecone", r)
+	return nil
+}
+
+// New returns an [ai.Retriever] that uses Pinecone.
 //
 // apiKey is the API key to use to access Pinecone.
 // If it is the empty string, it is read from the PINECONE_API_INDEX
```
Comment (ssbushi) [inline go/plugins/pinecone/genkit.go:54]:
```suggestion
// If it is the empty string, it is read from the PINECONE_API_KEY
```

**Comment by ianlancetaylor** [comment]
I've somehow broken this pull request by trying to merge other changes.  Will start again.

**Comment by ianlancetaylor** [comment]
Replaced by #276.  Sorry for the hassle.

**Pull request closed without merging**